watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -1186,8 +1229,43 @@ def rule_visc_k_phase(b, p):
     def _radius_stokes_comp(self):
         add_object_reference(self, "radius_stokes_comp", self.params.radius_stokes_comp)
 
+    def _molar_volume_comp(self):
+        add_object_reference(self, "molar_volume_comp", self.params.molar_volume_comp)
+
     def _diffus_phase_comp(self):
-        add_object_reference(self, "diffus_phase_comp", self.params.diffus_phase_comp)
+
+        if self.params.config.diffus_calculation == DiffusivityCalculation.none:
+            add_object_reference(
+                self, "diffus_phase_comp", self.params.diffus_phase_comp
+            )
+        else:
+            # self.params.config.diffus_calculation == DiffusivityCalculation.hayduklaudie
+            self.diffus_phase_comp = Var(
+                self.params.phase_list,
+                self.params.solute_set,
+                initialize=1e-9,
+                units=pyunits.m**2 * pyunits.s**-1,
+                doc="Molecular diffusivity of nonelectrolytes",
+            )
+
+            def rule_diffus_phase_comp(b, p, j):
+                # (Hayduk & Laudie, 1974)
+                diffus_coeff_inv_units = pyunits.s * pyunits.m**-2
+                visc_solvent_inv_units = pyunits.Pa**-1 * pyunits.s**-1
+                molar_volume_inv_units = pyunits.mol * pyunits.m**-3
+                # (1e# * name_inv_units) indicates unit conversions from H-L equation to base SI
+                return (b.diffus_phase_comp[p, j] * 1e4 * diffus_coeff_inv_units) * (
+                    (b.visc_d_phase[p] * 1e3 * visc_solvent_inv_units) ** 1.14
+                ) * (
+                    (b.molar_volume_comp[p, j] * 1e6 * molar_volume_inv_units) ** 0.589
+                ) == 13.26e-5
+
+            self.eq_diffus_phase_comp = Constraint(
+                self.params.phase_list,
+                self.params.solute_set,
+                rule=rule_diffus_phase_comp,
+            )
+        # TODO: add warning for use of ionic vs molecular diffusivity when ion+solute set > 1
```

Comment:
A few insights: 1) I doubt the units are treated accurately here.  A var already has a unit when it is created and it seems you multiplied the original unit with another unit.   Maybe using the unit conversion function is the correct way to go.  Also when a var has a power in the eq, its unit will carry the power too, so you may need to account for the units of all coefficient in the equation.  To assure everything is correct, you may wanna run an `assert_units_consistent` in the test.

Reply:
2) I quickly searched this Hayduk laudie equation you are implementing-  I saw some inconsistency in wrt the unit of "molar volume" or "molal volume" in different places, with units of cm^3/mol or cm^3/(mol g).  Would you make sure the accuracy of what is used ?   This equations seem to be an empirical correlation upon 87 substances (claimed in their original paper); would this mean you can only apply it for these 87 compounds? I am curious how useful is this method in practice.

Reply:
Overall, I would probably always encourage the use of a constant diffusion constant input at this time, but I wouldn't object having this method in place.  Just be very careful with the unit issue and precise documentation I think.

Reply:
@lbibl I added a dof and units check to the MCAS test in the latest commit. 

With the HL correlation, the coefficient units are effectively meaningless because the coefficient values were determined w.r.t. the specific units shown in the image here. I wanted to keep the hard coded coefficient values the same as the HL correlation for clarity.
![image](https://user-images.githubusercontent.com/101219154/217296796-0b7a86c0-8258-431c-9270-2a0ad0442553.png)
Therefore, I multiply the Var/Param (and units with the Var/Param) within the correlation by the inverse of the correct units and any necessary unit conversion to create dimensionless multiplicands and ensure the output is correct.

It's correct that only some molar volume values are tabulated but you can also estimate molar volume from the LaBas method using elements/functional groups of any species. This is actively being done in the pfas_treatment_analysis branch as ongoing work and the order of magnitude for the diffus_phase_comp is at least correct. The lack of data availability for PFAS species was the motivation for having a systematic method for estimating unknown diffusivities.

Reply:
@lbibl there's a new implementation if you can check it at some point

Reply:
@lbibl Another niche scenario that I'm curious of your input.

The GAC model is single species adsorption, denoted with a target_species argument in the unit model. In the past version when DiffusivityCalculation.HaydukLaudie, the diffus_phase_comp was only indexed with solute_set (and not ion_set). Therefore, background ions did not have a calculated diffusivity and did not require data. 

Now with accepting fixed and calculated by hl diffusivities when DiffusivityCalculation.HaydukLaudie, data **must** be provided for background compounds even though not required by the GAC model.

I added arbitrary diffusivities for the background components in the GAC tests. They are never used by the model but now required for the property package requirements to be fulfilled. I can see either just documenting this clearly or adding a warning that _no diffusivity data was provided for species  j, value is defaulted to 1e-5_ or something similar.